box/schema_user.go

```diff
@@ -0,0 +1,542 @@
+	// PrivilegeObjectTypeSequence - privilege type based on sequences.
+	// A sequence is a generator of ordered integer values.
+	PrivilegeObjectTypeSequence PrivilegeObjectType = "sequence"
+	// PrivilegeObjectTypeLuaEval - privilege type based on executing arbitrary Lua code.
+	PrivilegeObjectTypeLuaEval PrivilegeObjectType = "lua_eval"
+	// PrivilegeObjectTypeLuaCall - privilege type based on
+	// calling any global user-defined Lua function.
+	PrivilegeObjectTypeLuaCall PrivilegeObjectType = "lua_call"
+	// PrivilegeObjectTypeSQL - privilege type based on
+	// executing an arbitrary SQL expression.
+	PrivilegeObjectTypeSQL PrivilegeObjectType = "sql"
+)
+
+// Permission is a struct based on permission tarantool object
+// https://www.tarantool.io/en/doc/latest/admin/access_control/#permissions
+type Permission string
+
+const (
+	// PermissionRead allows reading data of the specified object.
+	// For example, this permission can be used to allow a user
+	// to select data from the specified space.
+	PermissionRead Permission = "read"
+	// PermissionWrite allows updating data of the specified object.
+	// For example, this permission can be used to allow
+	// a user to modify data in the specified space.
+	PermissionWrite Permission = "write"
+	// PermissionCreate allows creating objects of the specified type.
+	// For example, this permission can be used to allow a user to create new spaces.
+	// Note that this permission requires read and write access to certain system spaces.
+	PermissionCreate Permission = "create"
+	// PermissionAlter allows altering objects of the specified type.
+	// Note that this permission requires read and write access to certain system spaces.
+	PermissionAlter Permission = "alter"
+	// PermissionDrop allows dropping objects of the specified type.
+	// Note that this permission requires read and write access to certain system spaces.
+	PermissionDrop Permission = "drop"
+	// PermissionExecute for role,
+	// allows using the specified role. For other object types, allows calling a function.
+	// Can be used only for role, universe, function, lua_eval, lua_call, sql.
+	PermissionExecute Permission = "execute"
+	// PermissionSession allows a user to connect to an instance over IPROTO.
+	PermissionSession Permission = "session"
+	// PermissionUsage allows a user to use their privileges on database objects
+	// (for example, read, write, and alter spaces).
+	PermissionUsage Permission = "usage"
+)
+
+// Privilege is a structure that is used to create new rights,
+// as well as obtain information for rights.
+type Privilege struct {
+	// Permissions is a list of privileges that apply to the privileges object type.
+	Permissions []Permission
+	// ObjectType - one of privilege object types (it might be space,function, etc.).
+	ObjectType PrivilegeObjectType
+	// ObjectName - can be the name of a function or space,
+	// and can also be empty in case of universe access
+	ObjectName string
```

Comment:
And such...
```suggestion
	// Type - one of privilege object types (it might be space,function, etc.).
	Type PrivilegeObjectType
	// Name - can be the name of a function or space,
	// and can also be empty in case of universe access
	Name string
```

Reply:
Naming based on tarantool doc https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_schema/user_grant/#lua-function.box.schema.user.grant

Reply:
fixed